protocol/protobuf/proto/state_representation/space/joint/joint_state.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package state_representation.proto;
+
+import "state_representation/state.proto";
+
+message JointState {
+  State state = 1;
+  repeated string joint_names = 2;
```

Comment:
I can see why the fields here are of type `repeated`, but my question is if you repeat one field, are the other fields repeated the same number of times? In other words, could you **theoretically** create a message JointState where all those attributes have different length?

Reply:
Yes, there is no validation between repeated fields, and that's not something the protocol easily supports. However, the JointState constructor and equivalents will complain if the sizes don't match - so a malformed message won't be decoded correctly.

That reminds me that I intend to add another layer of exception handling, so that the `bool decode(const std::string&, T&)` is exception safe even for malformed messages.